metadata-ingestion/tests/unit/test_utilities.py

```diff
@@ -55,9 +55,297 @@ def test_groupby_unsorted():
 @pytest.mark.skipif(
     sys.version_info < (3, 7), reason="The LookML source requires Python 3.7+"
 )
-def test_default_sql_parser():
+def test_metadatasql_sql_parser_get_tables_from_simple_query():
     sql_query = "SELECT foo.a, foo.b, bar.c FROM foo JOIN bar ON (foo.a == bar.b);"
 
-    tables_list = DefaultSQLParser(sql_query).get_tables()
+    tables_list = MetadataSQLSQLParser(sql_query).get_tables()
+    tables_list.sort()
+    assert tables_list == ["bar", "foo"]
 
-    assert tables_list == ["foo", "bar"]
+
+@pytest.mark.integration
+@pytest.mark.skipif(
+    sys.version_info < (3, 7), reason="The LookML source requires Python 3.7+"
+)
+def test_sqllineage_sql_parser_get_tables_from_simple_query():
+    sql_query = "SELECT foo.a, foo.b, bar.c FROM foo JOIN bar ON (foo.a == bar.b);"
+
+    tables_list = MetadataSQLSQLParser(sql_query).get_tables()
+    tables_list.sort()
+    assert tables_list == ["bar", "foo"]
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(
+    sys.version_info < (3, 7), reason="The LookML source requires Python 3.7+"
+)
+def test_sqllineage_sql_parser_get_tables_from_complex_query():
+    sql_query = """
+(
+SELECT
+    CAST(substring(e, 1, 10) AS date) AS __d_a_t_e,
+    e AS e,
+    u AS u,
+    x,
+    c,
+    count(*)
+FROM
+    schema1.foo
+WHERE
+    datediff('day',
+    substring(e, 1, 10) :: date,
+    date :: date) <= 7
+    AND CAST(substring(e, 1, 10) AS date) >= date('2010-01-01')
+    AND CAST(substring(e, 1, 10) AS date) < getdate()
+GROUP BY
+    1,
+    2,
+    3,
+    4,
+    5)
+UNION ALL(
+SELECT
+CAST(substring(e, 1, 10) AS date) AS date,
+e AS e,
+u AS u,
+x,
+c,
+count(*)
+FROM
+schema2.bar
+WHERE
+datediff('day',
+substring(e, 1, 10) :: date,
+date :: date) <= 7
+    AND CAST(substring(e, 1, 10) AS date) >= date('2020-08-03')
+        AND CAST(substring(e, 1, 10) AS date) < getdate()
+    GROUP BY
+        1,
+        2,
+        3,
+        4,
+        5)
+"""
+
+    tables_list = SqlLineageSQLParser(sql_query).get_tables()
+    tables_list.sort()
+    assert tables_list == ["schema1.foo", "schema2.bar"]
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(
+    sys.version_info < (3, 7), reason="The LookML source requires Python 3.7+"
+)
+def test_sqllineage_sql_parser_get_columns_with_join():
+    sql_query = "SELECT foo.a, foo.b, bar.c FROM foo JOIN bar ON (foo.a == bar.b);"
+
+    columns_list = SqlLineageSQLParser(sql_query).get_columns()
+    columns_list.sort()
+    assert columns_list == ["a", "b", "c"]
+
+
+def test_sqllineage_sql_parser_get_columns_from_simple_query():
+    sql_query = "SELECT foo.a, foo.b FROM foo;"
+
+    columns_list = SqlLineageSQLParser(sql_query).get_columns()
+    columns_list.sort()
+    assert columns_list == ["a", "b"]
```

Comment:
Looks like there are only two kinds of tests, the ones testing get_tables and the ones testing get_columns, each type doing the same set of operations for different params. Consider refactoring with the `pytest.mark.parametrize` decorator - [examples](https://docs.pytest.org/en/6.2.x/example/parametrize.html#paramexamples)

Reply:
sure, will do, thanks.